test/e2e_node/remote/remote.go

```diff
@@ -146,6 +153,16 @@ func RunRemote(suite TestSuite, archive string, host string, cleanup bool, image
 		return "", false, fmt.Errorf("failed to create test result directory %q on host %q: %v output: %q", resultDir, host, err, output)
 	}
 
+	// We want to add credential provider flags here as it will work for both node conformance and node e2e
+	// credential provider binary and config is copied into workspace by the test suite, we need to provide
+	// config and binary location to kubelet via flags.
+	if installCredentialProvider {
+		var err error
+		if testArgs, err = addCredentialProviderFlags(testArgs, workspace); err != nil {
```

Comment:
It might be cleaner to use the KubeletConfiguration customization rather than adding kubelet flags.

It means that if someone wants to test with an outside credential provider they can provide a new config file with a path to where they installed their custom provider rather than needing to pass TEST_ARGS or similar.

Reply:
We can do with config as well but anyways we need to modify/add flags here as we know the location of workspace dir(where tar file is copied) at this point only, This might not be the issue with local runner but with remote runner we cannot know the location of credential-provider-bin and config before this point. 
The only way to directly use flag/config provided by user is the config and bin already installed in node image